docs/source/reference-testing.rst

```diff
@@ -219,3 +219,16 @@ Testing checkpoints
 
 .. autofunction:: assert_no_checkpoints
    :with:
+
+
+ExceptionGroup helpers
+----------------------
+
+.. autoclass:: RaisesGroup
+   :members:
+
+.. autoclass:: Matcher
+   :members:
+
+.. autoclass:: trio.testing._raises_group._ExceptionInfo
+   :members:
```

Comment:
These classes should have docstrings aimed at explaining their use to end-users, if we're exposing them publicly.

I'd also prefer to mark this as a provisional interface, and note that we hope for it to be obsoleted by upstream work in pytest itself someday soon.  (note: carefully noncommital about what happens or when)

Reply:
I wrote extensive docstrings... on a computer I'm currently 500km away from, and forgot to push. So I'll add them once back the 30th of december.
But good call to explicitly mark them as provisional

Reply:
Added them, and a note about them being provisional in the newsfragment.